SkyEditor.RomEditor.Rtdx.ConsoleApp/Scripts/Utilities/Rtdx/GenerateTranscript.csx

```diff
@@ -0,0 +1,67 @@
+// This script generates a text file containing all dialogues.
+// The resulting transcript.txt file will be saved in the current working directory.
+
+#load "../../../Stubs/RTDX.csx"
+
+using System;
+using System.IO;
+using SkyEditor.RomEditor.Domain.Rtdx.Constants;
+using SkyEditor.RomEditor.Domain.Rtdx;
+using SkyEditor.RomEditor.Domain.Rtdx.Structures;
+using System.Text.RegularExpressions;
+using System.Text;
+
+var rtdxRom = (RtdxRom) Rom;
+var usMessageBin = rtdxRom.GetUSMessageBin();
+var scriptStrings = new MessageBinEntry(rtdxRom.GetUSMessageBin().GetFile("script.bin"));
+
+string LuaScriptsPath = Path.Combine(rtdxRom.RomDirectory, "romfs", "Data", "StreamingAssets", "native_data", "script", "event");
+
+var transcript = new StringBuilder();
+
+foreach (var luaFile in new DirectoryInfo(LuaScriptsPath).GetFiles("*.lua", SearchOption.AllDirectories))
+{
+  transcript.AppendLine("------");
+  transcript.AppendLine(luaFile.Name);
+  transcript.AppendLine("------");
+
+  string[] lines = File.ReadAllLines(luaFile.FullName);
+  for (int i = 0; i < lines.Length; i++)
+  {
+    string line = lines[i];
+    var textIdMatch = Regex.Match(line, "TextID\\(\"(?<textIdValue>.+?)\",");
```

Comment:
It's fine because it's a script, but you could create a new instance of Regex with RegexOptions.Compiled to get a slight speed increase.